contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -244,12 +243,21 @@ contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions
       return (resultCode, callbackGasCostJuels);
     }
 
+    // Check that the subscription can still afford
+    if (
+      commitment.adminFee + costWithoutFulfillment + (juelsPerGas * SafeCast.toUint96(commitment.callbackGasLimit)) >
+      s_subscriptions[commitment.subscriptionId].balance
+    ) {
+      resultCode = 4; // IFunctionsRouter.FulfillResult.INSUFFICIENT_SUBSCRIPTION_BALANCE
```

Comment:
What happens afterwards? Can the funds be unlocked or is the user still charged?

Reply:
User is not charged in this scenario and eventually the request will time out.
We could make it that we charge them what we can, but then we would need to decide who gets what of the incomplete funds.
I don't believe this case should ever actually be hit unless something is going wrong, so if it is we need to alert on it.